sql/hive-thriftserver/v0.13.1/src/main/scala/org/apache/spark/sql/hive/thriftserver/Shim13.scala

```diff
@@ -154,8 +176,18 @@ private[hive] class SparkExecuteStatementOperation(
   }
 
   def run(): Unit = {
+    val sid = UUID.randomUUID().toString 
     logInfo(s"Running query '$statement'")
     setState(OperationState.RUNNING)
+    val group = hiveContext.sparkContext.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID) match {
+      case groupId: String =>
+        hiveContext.sparkContext.setJobDescription(statement)
+        groupId
+      case _ => hiveContext.sparkContext
+        .setJobGroup(parentSession.getSessionHandle.getSessionId.toString, statement)
+        sid
+    }
```

Comment:
Could be simplified to this:

```scala
val groupId = Option(hiveContext.sparkContext.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID)).getOrElse(sid)
hiveContext.sparkContext.setJobGroup(groupId, statement)
```

We shouldn't use session ID as group ID, otherwise all statements issued by the current session will be cancelled once the job group is cancelled.